File: typescript/examples/VaasExample/url_scan.ts
import {
  ClientCredentialsGrantAuthenticator,
  Vaas,
} from "gdata-vaas";

function throwError(errorMessage: string): never {
  throw new Error(errorMessage);
}

function getFromEnvironment(key: string) {
  return (
    process.env[key] ?? throwError(`Set ${key} in environment or .env file`)
  );
}

async function main() {
  const CLIENT_ID = getFromEnvironment("CLIENT_ID");
  const CLIENT_SECRET = getFromEnvironment("CLIENT_SECRET");
  const TOKEN_URL =
    "https://keycloak-vaas.gdatasecurity.de/realms/vaas/protocol/openid-connect/token";

  const authenticator = new ClientCredentialsGrantAuthenticator(
    CLIENT_ID,
    CLIENT_SECRET,
    TOKEN_URL
  );

  const vaas = new Vaas();
  const token = await authenticator.getToken()
  await vaas.connect(token);
  
  const url = new URL("https://secure.eicar.org/eicar.com");

  const verdict = await vaas.forUrl(url);
  console.log(verdict);
  vaas.close();
}

main().catch((e) => {
  console.log(e);
});

File: java/examples/VaasExample/src/main/java/de/gdata/vaasexample/UrlScan.java
package de.gdata.vaasexample;

import de.gdata.vaas.*;
import de.gdata.vaas.authentication.*;
import java.net.URI;
import java.net.URL;

public class UrlScan {
    public static void main(String[] args) throws Exception {
        var clientId = System.getenv("CLIENT_ID");
        var clientSecret = System.getenv("CLIENT_SECRET");
        var tokenUrl = System.getenv("TOKEN_URL");
        if (tokenUrl == null) {
            tokenUrl = "https://account.gdata.de/realms/vaas-production/protocol/openid-connect/token";
        }
        var vaasUrl = System.getenv("VAAS_URL");
        if (vaasUrl == null) {
            vaasUrl = "wss://gateway.production.vaas.gdatasecurity.de";
        } 

        var authenticator = new ClientCredentialsGrantAuthenticator(clientId, clientSecret, new URI(tokenUrl));
        var config = new VaasConfig(new URI(vaasUrl));
        var vaas = new Vaas(config, authenticator);
        var url = new URL("https://secure.eicar.org/eicar.com");
        var verdict = vaas.forUrl(url);
        System.out.printf("File %s was detected as %s", verdict.getSha256(), verdict.getVerdict());
    }
}